apps/server/src/shared/domain/entity/user.entity.ts

```diff
@@ -157,44 +157,51 @@ export class User extends BaseEntityWithTimestamps implements EntityWithSchool {
 			permissions = [...permissions, ...rolePermissions];
 		});
 
-		const setOfPermission = this.resolveSchoolPermissions(permissions, roles);
+		const setOfPermissions = this.resolveSchoolPermissions(permissions, roles);
 
-		const uniquePermissions = [...setOfPermission];
+		const uniquePermissions = [...setOfPermissions];
 
 		return uniquePermissions;
 	}
 
-	private resolveSchoolPermissions(permissions: string[], roles: Role[]) {
-		const setOfPermission = new Set(permissions);
+	private resolveSchoolPermissions(permissions: string[], roles: Role[]): Set<string> {
```

Comment:
Letztentlich wird die Methode dann zum User domain object wandern. Was erstmal ok ist solange der user selber darüber auskunft geben kann, oder es eine Zuweisung ist. Nur werden hier roles rein gegeben und mit diesen dann interagiert und informationen von diesen zurück geliefert. 

Desweiteren wollen wir eigentlich import { Configuration } from '@hpi-schul-cloud nur noch als Notlösung in den module configs lassen bis wir die hpi common komplett entfernen können. Der Rest soll eigentlich über die Nest Configuration laufen. 

Die Methode wurde von uns auch nur übernommen, aber sie hat sehr sehr viele if else etc.. daher die Trennung von der Orchestrierung und den Logik Operationen ist hier gar nicht mehr gewährleitet. Was es sehr schwer erfassbar macht was hier eigentlich passieren soll. 
Ich vermute ganz stark das wir das refactoren sollten, da wir es jetzt bereits zum 2. mal anfassen.
Das gerade in einem Bug Ticket machen wäre vermutlich etwas unglücklich.

Reply:
Naja, ich kann ja jetzt kein DO für den User in nem Bugfix einführen. Das wären ja riesige Änderungen. Generell sinnvoll fänd ich das aber definitiv.

Bezüglich Config: Man könnte perspektivisch dem DO-Constructor bzw. der Factory ein Config-Objekt übergeben. Dann müsste in der Klasse, die die Factory aufruft, der ConfigService injected werden, z.B. im Repo. Das hatte ich mir schon mal für die Schule überlegt, aber da war an anderer Stelle noch zu viel ungeklärt, um das zu implementieren.

Bezüglich Lesbarkeit: Ich hab die Logik noch ein bisschen in private Methoden aufgeteilt.

Reply:
Ahh stimmt bei der Factory gefällt mir. Egal ob direkt dort injected, oder übergeben als param hätte man darüber ein relative leicht zu realisierende Lösung.
Nein, nein kein DO einführen. :D 
Nur frage ich mich ob der Part des "rein" reichen der Rollen, Informationen rausziehen und diese return noch der Definintion entspricht das der User selber darüber auskunft geben kann. Ist für mich hart an der Grenze, nur bin ich mir nicht sicher ob vor, oder hinter dieser.

Reply:
Ich weiß es halt auch nicht sicher. Ich würde es jetzt in dem Ticket nicht ändern, nur beim Review ist mir aufgefallen das die Rollen von außen kommen und nicht bereits dem User bekannt sind. Was ok ist. Nur haben wir jetzt user.school auf der einen Seite, config eigentlich von außen da "theoretisch" nicht direkt user relevant, Dann kommt noch die Rolle von außen. Da bleibt dann gar nicht mehr soo viel "selbst Aussage".